### ISSUE TITLE
is it possible to set filter before render table?

### ISSUE THREAD

**Issue by antonyboom** [wenzhixin/bootstrap-table#2032]
I'm working with jstree library and loading my table by click on one of the child. My tree looks like 
Root > Parent [n] > Child[n]. So is it any way to set filter for Parent before initialize table? Parent in the tree has the same value in the table

This is my script:
```
$study.on('select_node.jstree', function (evt, data) {
        // ===== Initialize parent =====
        var loMainSelected = data;
        uiGetParents(loMainSelected);
        function uiGetParents(node) {
            try {
                var level = node.node.parents.length;
                var elem = $('#' + node.node.id);
                var parent = node.node.text;
                for (var ln = 0; ln <= level - 1; ln++) {
                    elem = elem.parent();
                    var child = elem.children()[2];
                    if (child != undefined) {
                        parent = child.text;
                    }
                }
                console.log(parent);
            }
            catch (err) {
                console.log('Error in uiGetParents');
            }
        }
        // ===== Click event on node =====
        for(var i = 0; i < data.selected.length; i++) {
            var node = data.instance.get_node(data.selected[i]).text;
            if (node == "Events") {
                $table.bootstrapTable(
                    'refresh', {
                        url: 'data/events.json'
                    });
            } else if (node == "Standards") {
                    load();
            }
        }
    })
    .jstree();
```

**Comment by djhvscf** [comment]
I think we need a fiddle in order to see the example working.. Can you provide a fiddle?? @antonyboom

**Comment by antonyboom** [comment]
https://jsfiddle.net/Lb1k2rw8/

the data is not loading to table but I guess my thought is clearly explained, thank you!

**Comment by antonyboom** [comment]
fixed fiddle 
https://jsfiddle.net/Lb1k2rw8/4/
so when I'm clicking on child1 I want to get only two rows for parent1 is it possible?

**Comment by wenzhixin** [comment]
It seems that your fiddle does not work. @antonyboom

**Comment by antonyboom** [comment]
it doesn't work from phone because it require double tap.
Double tap on root then double tap on parent[n] then double tap on child[1] and table will loaded. 
I have checked twice everything works fine in fiddle @wenzhixin

**Comment by djhvscf** [comment]
See this @antonyboom https://jsfiddle.net/djhvscf/Lb1k2rw8/9/

**Comment by antonyboom** [comment]
there is still show me same 3 rows without filtering. What does 'refreshOptions' do? @djhvscf

**Comment by djhvscf** [comment]
Ok.. but you should filter the array using the `jstree` functionality such as `get_node` or `get_parent`..
The refreshOptions initialize the table with a new object of options

**Comment by antonyboom** [comment]
OK, thank I'll try more... 
can you please answer on one more question? is it possible to header automatically? for example I have 4 json files all of them with different structure, can this library generate header automatically?

**Comment by djhvscf** [comment]
Let's use the `refreshMethod`

**Comment by antonyboom** [comment]
so it should be like in your example 
```
$(function () {
        $button.click(function () {
            $table.bootstrapTable('refresh');
        });
    });
```
but also inside function I have to describe columns filed and name?

**Comment by djhvscf** [comment]
No, Sorry!! You should use the `refreshOptions`

**Comment by antonyboom** [comment]
unfortunately this method doesn't... 
look fiddle https://jsfiddle.net/p9d2r77o/1/

**Comment by djhvscf** [comment]
See this one: https://jsfiddle.net/p9d2r77o/2/

**Comment by antonyboom** [comment]
yes this is exactly what I mean, header is the same, it doesn't load automatically, isn't ?

**Comment by djhvscf** [comment]
Aaaah ! Sorry now I understand what you mean. . Let me try again

**Comment by antonyboom** [comment]
sure take you time, thank you a lot for your support

**Comment by djhvscf** [comment]
What about this: https://jsfiddle.net/djhvscf/p9d2r77o/3/?

**Comment by antonyboom** [comment]
Awesome!! thanks a lot!

**Comment by djhvscf** [comment]
Great !
